mne_bids/write.py

```diff
@@ -278,8 +279,27 @@ def _participants_tsv(raw, subject_id, fname, overwrite=False,
                                   'list. Please set overwrite to '
                                   'True.' % subject_id)
         # otherwise add the new data
+        _old_keys = data.keys()
         data = _combine(orig_data, data, 'participant_id')
 
+        # loop through the keys in the original data to
+        # keep the original order, and add back
+        # the data in participants files that user added
+        for key in orig_data.keys():
+            # skip this key if mne-bids handles it
+            if key in _old_keys:
+                continue
+            # add them back per participant id
+            for p_id, val in zip(data['participant_id'], data[key]):
+                if p_id in orig_data['participant_id']:
+                    # get the participant row from orig and new data
+                    orig_p_idx = orig_data['participant_id'].index(p_id)
+                    p_idx = data['participant_id'].index(p_id)
+
+                    # assign original value to new data
+                    orig_val = orig_data[key][orig_p_idx]
+                    data[key][p_idx] = orig_val
```

Comment:
I feel this logic should go in a private function called `_combine_columns` and the old private method `_combine` should be renamed `_combine_rows`